libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Component that translates Solidity code into Yul at statement level and below.
+ */
+
+#include <libsolidity/codegen/ir/IRGeneratorForStatements.h>
+
+#include <libsolidity/codegen/ir/IRGenerationContext.h>
+#include <libsolidity/codegen/YulUtilFunctions.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::solidity;
+
+bool IRGeneratorForStatements::visit(VariableDeclarationStatement const& _varDeclStatement)
+{
+	for (auto const& decl: _varDeclStatement.declarations())
+		if (decl)
+			m_context.addLocalVariable(*decl);
+
+	solUnimplementedAssert(_varDeclStatement.declarations().size() == 1, "");
+	if (Expression const* expression = _varDeclStatement.initialValue())
+	{
+		expression->accept(*this);
+
+		solUnimplementedAssert(
+			*expression->annotation().type == *_varDeclStatement.declarations().front()->type(),
+			"Type conversion not yet implemented"
+		);
+		m_code <<
+			"let " <<
+			m_context.variableName(*_varDeclStatement.declarations().front()) <<
+			" := " <<
+			m_context.variable(*expression) <<
+			"\n";
+	}
+	else
+		for (auto const& decl: _varDeclStatement.declarations())
+			if (decl)
+				m_code << "let " << m_context.variableName(*decl) << "\n";
+
+	return false;
+}
+
+bool IRGeneratorForStatements::visit(Assignment const& _assignment)
+{
+	solUnimplementedAssert(_assignment.assignmentOperator() == Token::Assign, "");
+
+	_assignment.rightHandSide().accept(*this);
+
+	solUnimplementedAssert(
+		*_assignment.rightHandSide().annotation().type == *_assignment.leftHandSide().annotation().type,
+		"Type conversion not yet implemented"
+	);
+	// TODO proper lvalue handling
+	auto const& identifier = dynamic_cast<Identifier const&>(_assignment.leftHandSide());
```

Comment:
Shouldn't this be a `solUnimplementedAssert`?

Reply:
I'm not sure - these dynamic casts will throw exceptions anyways, if they fail, and we have the ``TODO``s - do you think we should require this code to have proper assertions everywhere already?

Reply:
Ok, I guess it's fine without.

Reply:
It's only here I think and this function will need a lot of work anyways, so yeah, I think it's fine.

Reply:
Ah, ok... the ``The function can very well return multiple vars.`` TODO's don't assert as well... still, I'd leave it at that for this PR.